barretenberg/cpp/src/barretenberg/relations/logderiv_lookup_relation.hpp

```diff
@@ -0,0 +1,197 @@
+#pragma once
+#include <array>
+#include <tuple>
+
+#include "barretenberg/common/constexpr_utils.hpp"
+#include "barretenberg/honk/proof_system/logderivative_library.hpp"
+#include "barretenberg/polynomials/polynomial.hpp"
+#include "barretenberg/polynomials/univariate.hpp"
+#include "barretenberg/relations/relation_types.hpp"
+
+namespace bb {
+
+template <typename FF_> class LogDerivLookupRelationImpl {
+  public:
+    using FF = FF_;
+    static constexpr size_t READ_TERMS = 1;
+    static constexpr size_t WRITE_TERMS = 1;
+    // 1 + polynomial degree of this relation
+    static constexpr size_t LENGTH = 5; // both subrelations are degree 4
+
+    static constexpr std::array<size_t, 2> SUBRELATION_PARTIAL_LENGTHS{
+        LENGTH, // inverse construction sub-relation
+        LENGTH  // log derivative lookup argument sub-relation
+    };
+
+    // WORKTODO: shouldnt first one be 2? Why do tests pass with 1? (Note: if setting to 2, both need to be 2 due to
+    // structure of relation algebra)
+    static constexpr std::array<size_t, 2> TOTAL_LENGTH_ADJUSTMENTS{
+        1, // inverse construction sub-relation
+        1  // log derivative lookup argument sub-relation
+    };
+
+    static constexpr std::array<bool, 2> SUBRELATION_LINEARLY_INDEPENDENT = { true, false };
+
+    template <typename AllEntities> inline static bool skip(const AllEntities& in)
+    {
+        // Ensure the input does not contain a lookup gate or data that is being read
+        return in.q_lookup.is_zero() && in.lookup_read_counts.is_zero();
+    }
+
+    // Used to determine whether the polynomial of inverses must be computed at a given row
+    template <typename AllValues> static bool operation_exists_at_row(const AllValues& row)
+    {
+        // is lookup gate or row contains data that has been read
```

Comment:
Can you make it clearer that if data has been read already we don't have to compute the inverse because it's already been computed? I was a bit confused.

Reply:
Tried to improve the comments a bit.. I think its clearer now